MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,920 @@ If the requested :code:`timeout_nsec` exceeds
+large enough, the return code is :code:`C.YDB_ERR_INVSTRLEN` and
+:code:`str.GetLenUsed()` returns the length required (in this case,
+the value returned by :code:`str.GetLenUsed()` will be greater than
+that returned by :code:`str.GetLenAlloc()` until corrected by
+application code.
+
+Note that the length of a string in `zwrite format`_ is always less
+than or equal to the string in its original, unencoded format.
+
+----------------------------------
+Go Simple API BufferTArray Methods
+----------------------------------
+
+Go DeleteExclS()
+----------------
+
+.. code-block:: go
+
+    DeleteExclS() int
+
+:code:`DeleteExclS()` wraps `ydb_delete_excl_s()`_ to delete all local
+variable trees except those of the local variables whose names are
+specified in the :code:`BufferTArray` structure with a return code of
+:code:`C.YDB_OK`. In the special case where the :code:`elemsUsed` of
+the :code:`BufferT` is zero, it deletes all local variable trees. If
+your application mixes M and Go code, read and understand the warning
+in the description of `ydb_delete_excl_s()`_.
+
+In the unlikely event that the :code:`elemsUsed` of the
+:code:`BufferT` structure exceeds :code:`C.YDB_MAX_NAMES`, the return
+code is :code:`C.YDB_ERRNAMECOUNT2HI`. Any other error will result in
+an `error return code`_ prefixed with :code:`C.`.
+
+Go TpS()
+--------
+
+.. code-block:: go
+
+    TpS(tpfn unsafe pointer, tpfnparm unsafe.pointer, transid *string) int
+
+:code:`TpS()` wraps `ydb_tp_s()`_ to implement `Transaction
+Processing`_. :code:`tpfn` is a pointer to a C function with one
+parameter, a pointer to an arbitrary data structure in YottaDB address
+space. Please see both the description of `ydb_tp_s()`_ and the
+section on `Transaction Processing`_ for details.
+
+Since Go does not permit a pointer to a Go function to be passed as a
+parameter, :code:`tpfn` is a pointer to a C function that implements
+logic for the transaction. For a Go application, the C function is an
+interlude that in turn calls the Go function. YottaDB provides a shell
+script `GenYDBInterlude.sh`_ to generate interlude functions.
+
+A function implementing logic for a transaction should return one of
+the following:
+
+- :code:`C.YDB_OK` to indicate that per application logic, the
+  transaction can be committed. The YottaDB database engine will
+  commit the transaction if it is able to, as discussed in
+  `Transaction Processing`_, and if not, will call the function again.
+- :code:`C.YDB_TPRESTART` to indicate that the transaction should
+  restart, either because application logic has so determined or
+  because a YottaDB function called by the function has returned
+  :code:`C.YDB_TPRESTART`.
+- :code:`C.YDB_ROLLBACK` to indicate that :code:`TpS()` should not
+  commit the transaction.
+
+Also, in order to provide the function implementing the transaction
+logic with a parameter or parameters, :code:`tpfnparm` is passed to
```

Comment:
Note tpfnparm should (probably? need expert advice on this that isn't me as we are emphatically exploring the limitations of cgo) be allocated as C memory as the parameter is passed back to Go so could be garbage collected. This would be a prime question for our reviewer(s).